libtransmission/makemeta.cc

```diff
@@ -319,8 +320,18 @@ static uint8_t* getHashInfo(tr_metainfo_builder* b)
 
         TR_ASSERT(bufptr - buf == (int)thisPieceSize);
         TR_ASSERT(leftInPiece == 0);
-        tr_sha1(walk, buf, (int)thisPieceSize, nullptr);
-        walk += SHA_DIGEST_LENGTH;
+        auto const digest = tr_sha1(std::string_view{ buf, thisPieceSize });
+        if (!digest)
+        {
+            b->my_errno = errno;
```

Comment:
I couldn't find any `E...` that really applies. Maybe it would make sense to add a new `tr_metainfo_builder_err` enum item for this instead?

Reply:
Best I can come up with is `EIO`. Not sure how frequent we'll see this error in reality, but I guess there's no such thing as too much error details so a new enum item is not the worst idea.

Reply:
New error enum added in 4f0d0f386c3fad6a76c7ee588cf4f6dccf14110d

Reply:
OTOH, error message itself might be enough...

Reply:
I don't feel strongly about it either way.

I have a slight preference for setting a different errno instead of making a new enum item just so that I don't have to go and check all the clients' error handling code though :)

Reverted back + `EIO` in bb4129bed2b55381a2f8911c6ccdebc26de9616e